src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -197,7 +197,38 @@ public async Task<string> SubmitPledge(CreatePostRequest request)
 
             return _encodingService.Encode(pledgeId, EncodingType.PledgeId);
         }
+        public async Task RejectApplications(RejectApplicationsPostRequest request)
+        { 
+            var rejectApplicationsRequest = new RejectApplicationsRequest
+            {
+                ApplicationsToReject = request.ApplicationsToReject,
+                UserId = _userService.GetUserId(),
+                UserDisplayName = _userService.GetUserDisplayName(),
+            };
+
+            await _pledgeService.RejectApplications(rejectApplicationsRequest, request.AccountId, request.PledgeId);
+        }
+
+        public async Task<RejectApplicationsViewModel> GetRejectApplicationsViewModel(ApplicationsPostRequest request)
+        {
+            var rejectApplicationsViewModel = new RejectApplicationsViewModel
+            {
+                ApplicationsToReject = request.ApplicationsToReject,
+                EncodedAccountId = request.EncodedAccountId,
+                EncodedPledgeId = request.EncodedPledgeId
+            };
 
+            foreach (var applicationId in request.ApplicationsToReject) 
+            {
+                var application = await _pledgeService.GetApplication(request.AccountId, request.PledgeId, applicationId);
```

Comment:
This means we hit the api once for each application marked for rejection, which is not great for performance. In addition, it violates the apim pattern we have whereby each action in the UI has an outer api endpoint matching the route and parameters (albeit no encoding).

Therefore we need to remove this in favour of an endpoint that will give us just the data we need for this page - a list of Ids and names, at a glance - and nothing else.

Reply:
Yes sure. I have created a new end-point that gives us just the data we need for this view. I have refactored this code.